internal/ebiten/sound.go

```
@@ -0,0 +1,175 @@
+package ebiten
+
+import (
+	"bytes"
+	"math"
+	"math/rand"
+
+	"github.com/hajimehoshi/ebiten/v2/audio"
+	"github.com/hajimehoshi/ebiten/v2/audio/vorbis"
+)
+
+// Sound is an ebiten implementation of engine.Sound.
+type Sound struct {
+	group   string
+	options [][]byte
+	context *audio.Context
+	player  *audio.Player
+
+	sc     *SoundControl
+	volume float64
+}
+
+// Play implements the Play method of engine.Sound.
+func (s *Sound) Play() {
+
+	if len(s.options) == 0 {
+		return
+	}
+
+	if s.player != nil {
+		s.player.Close()
+	}
+
+	option := s.options[rand.Intn(len(s.options))]
+	sound, _ := vorbis.Decode(s.context, bytes.NewReader(option))
+
+	// NOTE we may want to cache the players
+	s.player, _ = audio.NewPlayer(
```

spotted an ignored error ;)